File: app/src/main/java/com/commonpepper/photosen/ui/activities/CategoriesActivity.java
package com.commonpepper.photosen.ui.activities;

import android.os.Bundle;

import com.commonpepper.photosen.R;
import com.commonpepper.photosen.ui.adapters.CategoriesAdapter;
import com.google.android.flexbox.AlignItems;
import com.google.android.flexbox.FlexDirection;
import com.google.android.flexbox.FlexWrap;
import com.google.android.flexbox.FlexboxLayoutManager;
import com.google.android.flexbox.JustifyContent;
import com.google.android.material.navigation.NavigationView;

import androidx.annotation.Nullable;
import androidx.appcompat.widget.Toolbar;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

public class CategoriesActivity extends AbstractNavActivity {
    Toolbar mToolbar;
    RecyclerView mRecyclerView;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_categories);

        mToolbar = findViewById(R.id.toolbar);
        drawerLayout = findViewById(R.id.drawer_layout);
        NavigationView navigationView = findViewById(R.id.nav_view);
        mRecyclerView = findViewById(R.id.categories_recycler);

        setSupportActionBar(mToolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setDisplayShowHomeEnabled(true);
        getSupportActionBar().setTitle(getString(R.string.categories));

        navigationView.getMenu().findItem(R.id.drawer_categories).setCheckable(true);
        navigationView.getMenu().findItem(R.id.drawer_categories).setChecked(true);
        navigationView.setNavigationItemSelectedListener(this);

        mRecyclerView.setHasFixedSize(true);
        mRecyclerView.setLayoutManager(new GridLayoutManager(this, 2));
        mRecyclerView.setAdapter(new CategoriesAdapter(getResources().getStringArray(R.array.categories)));
    }

    @Override
    public boolean onSupportNavigateUp() {
        onBackPressed();
        return true;
    }
}

File: app/src/main/java/com/commonpepper/photosen/ui/activities/CategoriesActivity.kt
package com.commonpepper.photosen.ui.activities

import android.os.Bundle
import androidx.drawerlayout.widget.DrawerLayout
import androidx.recyclerview.widget.GridLayoutManager
import com.commonpepper.photosen.R.*
import com.commonpepper.photosen.ui.adapters.CategoriesAdapter
import kotlinx.android.synthetic.main.activity_categories.*
import kotlinx.android.synthetic.main.navigation_view.*

class CategoriesActivity : AbstractNavActivity() {
    override val abstractDrawerLayout: DrawerLayout get() = drawerLayout

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(layout.activity_categories)
        setSupportActionBar(toolbar)
        supportActionBar?.apply {
            setDisplayHomeAsUpEnabled(true)
            setDisplayShowHomeEnabled(true)
            title = getString(string.categories)
        }
        navigationView.menu.findItem(id.drawer_categories).apply {
            isCheckable = true
            isChecked = true
        }
        navigationView.setNavigationItemSelectedListener(this)
        recyclerCategories.setHasFixedSize(true)
        recyclerCategories.layoutManager = GridLayoutManager(this, 2)
        recyclerCategories.adapter = CategoriesAdapter(resources.getStringArray(array.categories))
    }

    override fun onSupportNavigateUp(): Boolean {
        onBackPressed()
        return true
    }
}
